From 755daf78769b048c868e6b62c68b4e545494c181 Mon Sep 17 00:00:00 2001
From: Stephen Shkardoon <ss23@ss23.geek.nz>
Date: Thu, 21 Jun 2018 14:38:30 +1200
Subject: [PATCH] Modify configure to prevent unnecessary kqueue checks

By only running the TORRENT_CHECK_KQUEUE_SOCKET_ONLY check if kqueue support
is already detected, we increase the number of platforms that we can
cross compile on.
Otherwise, the cross compilation fails due to TORRENT_CHECK_KQUEUE_SOCKET_ONLY
using AC_RUN_IFELSE, which fails during cross compilation.
---
 scripts/checks.m4 | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/scripts/checks.m4 b/scripts/checks.m4
index 8d77fc5e..c9333561 100644
--- a/scripts/checks.m4
+++ b/scripts/checks.m4
@@ -88,6 +88,7 @@ AC_DEFUN([TORRENT_CHECK_KQUEUE], [
     [
       AC_DEFINE(USE_KQUEUE, 1, Use kqueue.)
       AC_MSG_RESULT(yes)
+      TORRENT_CHECK_KQUEUE_SOCKET_ONLY
     ], [
       AC_MSG_RESULT(no)
     ])
@@ -137,7 +138,6 @@ AC_DEFUN([TORRENT_WITH_KQUEUE], [
     [
         if test "$withval" = "yes"; then
           TORRENT_CHECK_KQUEUE
-          TORRENT_CHECK_KQUEUE_SOCKET_ONLY
         fi
     ])
 ])
@@ -149,11 +149,9 @@ AC_DEFUN([TORRENT_WITHOUT_KQUEUE], [
     [
       if test "$withval" = "yes"; then
         TORRENT_CHECK_KQUEUE
-        TORRENT_CHECK_KQUEUE_SOCKET_ONLY
       fi
     ], [
         TORRENT_CHECK_KQUEUE
-        TORRENT_CHECK_KQUEUE_SOCKET_ONLY
     ])
 ])
 
